File: frontend/src/app/modules/user/model/UserProfileDTO.ts
export class UserProfileDTO {
  name: string;
  username: string;
  email: string;
  photo: string;
  birthdate: Date | null = null;

  constructor(
    name: string,
    username: string,
    email: string,
    photo: string,
    birthdate: Date | null
  ) {
    this.name = name;
    this.username = username;
    this.email = email;
    this.photo = photo;
    this.birthdate = birthdate;
  }
}

File: backend/src/main/java/com/orange/Crisalis/dto/UserProfileDTO.java
package com.orange.Crisalis.dto;

import com.orange.Crisalis.security.Entity.UserEntity;
import lombok.*;

import java.util.Date;


@Data
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class UserProfileDTO {
    private String name;
    private String username;
    private String email;
    private String photo;
    private Date birthdate;

    public UserProfileDTO(UserEntity userEntity){
        this.name = userEntity.getName();
        this.username = userEntity.getUsername();
        this.email = userEntity.getEmail();
        this.photo = userEntity.getPhoto();
        this.birthdate = userEntity.getBirthdate();
    }
}
